packages/debugger-extension/src/index.ts

```diff
@@ -695,13 +696,34 @@ const main: JupyterFrontEndPlugin<void> = {
       }
     });
 
+    commands.addCommand(CommandIDs.pause, {
+      label: trans.__('Enable / Disable pausing on exceptions'),
+      caption: trans.__('Enable / Disable pausing on exceptions'),
+      className: 'jp-PauseOnExceptions',
+      icon: Debugger.Icons.pauseOnExceptionsIcon,
+      isToggled: () => {
+        return service.isPausingOnExceptions;
+      },
+      isEnabled: () => {
+        return !!service.isStarted;
```

Comment:
What about kernel not supporting that type of exceptions?

Reply:
At worst, I think the kernel would just run until it hit the the exception instead of breaking immediately before. That being said, pauseOnExceptions sets the breakpoint argument options via the debug protocol so my assumption is that it should be kernel agnostic assuming their is a debug protocol for it. I have only tried running it with the ipython and xeus python kernels though.

Reply:
I understand but the debug protocol does specify an exchange of capabilities at start-up. Example for ipykernel (server reply):

![image](https://user-images.githubusercontent.com/8435071/148030507-e12a1fdc-75e7-43a9-8f75-e5e58fbe250c.png)

So it seems, we do have a way to determine if the kernel supports it.

Reply:
Those are currently used to build the options object that gets sent to debugger. So if none are defined for a kernel then pause on exception would never pause. Otherwise it should stop on those breakpoint filters.

Reply:
From the user point of view, this means the button will not do anything what should be avoided. This is why it will be nice if the `isEnabled` or perhaps the `isVisible` attribute of the command will be set accordingly.